src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBrokerHelper.cs

```diff
@@ -0,0 +1,488 @@
+                // intent. Activity needs to be launched from calling app
+                // to get the calling app's metadata if needed at BrokerActivity.
+
+                Bundle addAccountOptions = new Bundle();
+                addAccountOptions.PutString(BrokerConstants.BrokerAccountManagerOperationKey, BrokerConstants.GetIntentForInteractiveRequest);
+
+                _logger.Info("Broker Account Name: " + GetValueFromBrokerPayload(brokerPayload, BrokerParameter.Username));
+
+                result = _androidAccountManager.AddAccount(BrokerConstants.BrokerAccountType,
+                    BrokerConstants.AuthtokenType, null, addAccountOptions, null,
+                    null, GetPreferredLooper(callerActivity));
+
+                if (result == null)
+                {
+                    _logger.Info("Android account manager AddAccount didn't return any results. ");
+                }
+
+                Bundle bundleResult = (Bundle)result?.Result;
+                // Broker should throw OperationCanceledException if token is not available
+                intent = (Intent)bundleResult?.GetParcelable(AccountManager.KeyIntent);
+
+                //Validate that the intent was created succsesfully.
+                if (intent != null)
+                {
+                    _logger.Info("Intent created from BundleResult is not null. ");
+                    // Need caller info UID for broker communication
+                    intent.PutExtra(BrokerConstants.CallerInfoUID, Binder.CallingUid);
+                }
+                else
+                {
+                    _logger.Info("Intent created from BundleResult is null. ");
+                    throw new MsalException(MsalError.NullIntentReturnedFromBroker, MsalErrorMessage.NullIntentReturnedFromBroker);
+                }
+
+                intent = GetInteractiveBrokerIntent(brokerPayload, intent);
+            }
+            catch
+            {
+                _logger.Error("Error when trying to acquire intent for broker authentication.");
+                throw;
+            }
+
+            return intent;
+        }
+
+        public string GetAuthTokenSilently(IDictionary<string, string> brokerPayload, Activity callerActivity)
+        {
+            GetBrokerAccountInfo(brokerPayload, callerActivity);
+            Bundle silentOperationBundle = GetSilentBrokerBundle(brokerPayload);
+            silentOperationBundle.PutString(BrokerConstants.BrokerAccountManagerOperationKey, BrokerConstants.AcquireTokenSilent);
+
+            _logger.Info("Broker Account Name: " + GetValueFromBrokerPayload(brokerPayload, BrokerParameter.Username));
+
+            var result = _androidAccountManager.AddAccount(BrokerConstants.BrokerAccountType,
+                BrokerConstants.AuthtokenType, null, silentOperationBundle, null,
+                null, GetPreferredLooper(callerActivity));
+
+            if (result == null)
```

Comment:
this looks similar to the interactive code above...can any of it be reused in a separate method?

Reply:
it is similar but not enough to be extracted to a separate method